spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/RabbitConnectionFactoryBean.java

```diff
@@ -629,4 +627,36 @@ protected SSLContext createSSLContext() throws NoSuchAlgorithmException {
 		return SSLContext.getInstance(this.sslAlgorithm);
 	}
 
+
+	/**
+	 * In case the client doesn't provide the sslProperties location or the truststore is null delegate to Java's default
+	 * trust mechanism
+	 * @throws NoSuchAlgorithmException
+	 * @throws KeyManagementException
+	 * @throws KeyStoreException
+     */
+	private void useDefaultTrustStoreMechanism() throws NoSuchAlgorithmException, KeyManagementException,
+			KeyStoreException {
+		SSLContext c = SSLContext.getInstance(this.sslAlgorithm);
+		c.init(null, new TrustManager[] {  getDefaultTrustManager(TrustManagerFactory.getDefaultAlgorithm(), null) }, null);
+		this.connectionFactory.useSslProtocol(c);
+	}
+
+
+	private static X509TrustManager getDefaultTrustManager(String algorithm, KeyStore keystore) throws NoSuchAlgorithmException, KeyStoreException {
+
+		TrustManagerFactory factory = TrustManagerFactory.getInstance(algorithm);
+		X509TrustManager x509TrustManager = null;
+        factory.init(keystore);
```

Comment:
Thank you for some info, @harrydevnull .

I don't mind that `TrustManagerFactory.getDefaultAlgorithm()`, I just worry why we don't use it like this:
```
ctx.init(null, factory.getTrustManagers(), null);
```
Why do you need to extract exactly one `x509TrustManager` and don't use all of them for the `SSLContext`?